java/java/src/main/java/org/whispersystems/libsignal/groups/GroupSessionBuilder.java

```diff
@@ -33,14 +36,14 @@ public GroupSessionBuilder(SenderKeyStore senderKeyStore) {
   }
 
   /**
-   * Construct a group session for receiving messages from senderKeyName.
+   * Construct a group session for receiving messages from sender.
    *
-   * @param senderKeyName The (groupId, senderId, deviceId) tuple associated with the SenderKeyDistributionMessage.
+   * @param sender The address of the device that sent the message.
    * @param senderKeyDistributionMessage A received SenderKeyDistributionMessage.
    */
-  public void process(SenderKeyName senderKeyName, SenderKeyDistributionMessage senderKeyDistributionMessage) {
+  public void process(SignalProtocolAddress sender, SenderKeyDistributionMessage senderKeyDistributionMessage) {
     synchronized (GroupCipher.LOCK) {
```

Comment:
Unrelated to this PR, but should we remove this lock?

Reply:
Probably, yes. cc @greyson-signal

Reply:
Yeah, please no extra locks, we can wrap it in our own locking structure. Just annotate the class as not thread-safe, similar to what I did in `SessionCipher`.